lib/rubocop/cop/lint/unneeded_splat_expansion.rb

```diff
@@ -80,12 +78,36 @@ def on_splat(node)
 
         private
 
+        def unneeded_splat_expansion(node)
+          literal_expansion?(node) do |object|
+            if object.send_type?
+              return if array_new_inside_array_literal?(object)
+
+              grandparent = node.parent.parent
+              return if grandparent &&
+                        !ASSIGNMENT_TYPES.include?(grandparent.type)
+            end
+
+            yield
+          end
+        end
+
+        def array_new_inside_array_literal?(array_new_node)
+          return false unless array_new?(array_new_node)
+
+          grandparent = array_new_node.parent.parent
+          grandparent.array_type? && grandparent.children.count > 1
+        end
+
+        # rubocop:disable Metrics/AbcSize
```

Comment:
Hm. This actually exceeds the ABC size (which is already inflated) of any other method in the entire application. We should probably think about whether this method can be simplified, extracted, or factored differently. 🙂

Reply:
Thanks!

That's a tough one. I'm not very happy with my current solution, because we have to use `corrector.replace(loc.operator, '')` instead of `corrector.remove(loc.operator)`...